ksqldb-api-client/src/main/java/io/confluent/ksql/api/client/impl/QueryResponseHandler.java

```diff
@@ -35,10 +52,32 @@
 
   @Override
   protected void doHandleBodyBuffer(final Buffer buff) {
+    final Buffer strippedValidJson = toJsonMsg(buff, true);
+
+    if (strippedValidJson.toString().equals("")) {
+      return;
+    }
+
     if (!hasReadArguments) {
-      handleArgs(buff);
+      handleArgs(strippedValidJson);
     } else {
-      handleRow(buff);
+      final JsonNode rowNode;
+      try {
+        rowNode = JSON_MAPPER.readTree(strippedValidJson.getBytes());
+      } catch (Exception e) {
+        cf.completeExceptionally(e);
+        return;
+      }
+
+      if (rowNode.has(FINAL_MESSAGE_KEY)) {
+        doHandleBodyEnd();
+      } else if (rowNode.has(ERROR_MESSAGE_KEY)) {
+        final KsqlException exception = new KsqlException(
```

Comment:
Before, I believe this would be handled by calling the following rather than failing the request:

```
queryResult.handleError(new KsqlException(
            jsonObject.getString("message")
        ));
```

We should try to keep consistent behavior there.